sfge/src/main/java/com/salesforce/cli/CacheCreator.java

```diff
@@ -0,0 +1,112 @@
+package com.salesforce.cli;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import com.salesforce.config.SfgeConfig;
+import com.salesforce.config.SfgeConfigProvider;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/** Creates cache data that can be used to execute Delta runs. */
+public final class CacheCreator {
+    private static final Logger LOGGER = LogManager.getLogger(CacheCreator.class);
+
+    private Dependencies dependencies;
+
+    public CacheCreator() {
+        this.dependencies = new Dependencies();
+    }
+
+    CacheCreator(Dependencies dependencies) {
+        this.dependencies = dependencies;
+    }
+
+    /**
+     * Creates cache data that can be used for future delta runs
+     *
+     * @param result from the current execution
+     */
+    public void create(Result result) {
+        createFilesToEntriesCache(result.getFilesToEntriesMap());
+    }
+
+    private void createFilesToEntriesCache(FilesToEntriesMap filesToEntriesMap) {
+        final String jsonData = filesToEntriesMap.toJsonString();
+        final String filesToEntriesDataFilename = getFilesToEntriesDataFilename();
+
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.info(
+                    "About to create cache for filesToEntriesMap at " + filesToEntriesDataFilename);
+        }
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(
+                    "Content to be written in " + filesToEntriesDataFilename + ": " + jsonData);
+        }
+
+        try {
+            dependencies.writeFile(filesToEntriesDataFilename, jsonData);
+        } catch (IOException e) {
+            // Putting a warning instead of stopping the process since this is not in the critical
+            // path
+            if (LOGGER.isWarnEnabled()) {
+                LOGGER.warn("Exception occurred while writing filesToEntriesMap cache data.", e);
+            }
+        }
+    }
+
+    private String getFilesToEntriesDataFilename() {
+        final SfgeConfig sfgeConfig = SfgeConfigProvider.get();
+        final String cacheDir = sfgeConfig.getCacheDir();
```

Comment:
I'd prefer to not store it in the current directory so that it stays hidden unless specifically looked for. Agree that a single config variable would be sleeker.

Reply:
@jag-j I just remembered why I put the directory in a different config. If/When you add more data to cache (during Phase 2) such as Graph state, you may want to reuse the same directory and continue to have control over the names of files within.